sql: RETURNING causes pgwire panic

Adding a test like this in TestPGPreparedExec
```
466         {
467             "INSERT INTO d.T VALUES ($1) RETURNING 1",
468             []preparedExecTest{
469                 base.Params(1).RowsAffected(1),
470             },
471         },
```

Causes:

```
panic: runtime error: index out of range

goroutine 21058 [running]:
panic(0x160bfc0, 0xc82000e110)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).sendResponse(0xc8201841c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc8208f2a80, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:666 +0xd92
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).executeStatements(0xc8201841c0, 0xc8201862b6, 0x27, 0xc820a8f240, 0x1, 0x1, 0x2578ba8, 0x0, 0x0, 0x0, ...)
```

@mjibson

--- a/sql/delete.go
+++ b/sql/delete.go
@@ -55,6 +55,7 @@ func (p *planner) Delete(n *parser.Delete, autoCommit bool) (planNode, *roachpb.
 	}
 
 	if p.prepareOnly {
+		// Return the result column types.
 		return rh.getResults(), nil
 	}
 

--- a/sql/insert.go
+++ b/sql/insert.go
@@ -246,7 +246,8 @@ func (p *planner) Insert(n *parser.Insert, autoCommit bool) (planNode, *roachpb.
 	}
 
 	if p.prepareOnly {
-		return nil, nil
+		// Return the result column types.
+		return rh.getResults(), nil
 	}
 
 	if isSystemConfigID(tableDesc.GetID()) {

--- a/sql/pgwire_test.go
+++ b/sql/pgwire_test.go
@@ -298,7 +298,7 @@ func TestPGPreparedQuery(t *testing.T) {
 			base.Results("username", "STRING", false, sql.NullBool{}),
 		},
 		"SHOW DATABASES": {
-			base.Results("system"),
+			base.Results("d"),
 		},
 		"SHOW GRANTS ON system.users": {
 			base.Results("users", "root", "DELETE,GRANT,INSERT,SELECT,UPDATE"),
@@ -316,6 +316,20 @@ func TestPGPreparedQuery(t *testing.T) {
 			base.Params(1).Results(2),
 		},
 		// TODO(mjibson): test date/time types
+
+		"INSERT INTO d.T VALUES ($1) RETURNING 1": {
+			base.Params(1).Results(1),
+		},
+		/* TODO(mjibson): fix #4658
+		"INSERT INTO d.T VALUES ($1) RETURNING $1": {
+			base.Params(1).Results(1),
+			base.Params(3).Results(3),
+		},
+		"INSERT INTO d.T VALUES ($1) RETURNING $1, 1 + $1": {
+			base.Params(1).Results(1, 2),
+			base.Params(3).Results(3, 4),
+		},
+		*/
 	}
 
 	s := server.StartTestServer(t)
@@ -364,6 +378,10 @@ func TestPGPreparedQuery(t *testing.T) {
 				}
 			}
 		}
+	}
+
+	if _, err := db.Exec(`CREATE DATABASE d; CREATE TABLE d.t (a INT)`); err != nil {
+		t.Fatal(err)
 	}
 
 	for query, tests := range queryTests {

--- a/sql/update.go
+++ b/sql/update.go
@@ -166,6 +166,7 @@ func (p *planner) Update(n *parser.Update, autoCommit bool) (planNode, *roachpb.
 				return nil, roachpb.NewError(err)
 			}
 		}
+		// Return the result column types.
 		return rh.getResults(), nil
 	}
 

